File: rb/features.rb
# WeUltrarich SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WeUltrarichFeatures
  def self.make_feature(name)
    case name
    when "base"
      WeUltrarichBaseFeature.new
    when "ratelimit"
      WeUltrarichRatelimitFeature.new
    when "retry"
      WeUltrarichRetryFeature.new
    when "test"
      WeUltrarichTestFeature.new
    when "timeout"
      WeUltrarichTimeoutFeature.new
    else
      WeUltrarichBaseFeature.new
    end
  end
end

File: py/features.py
# WeUltrarich SDK feature factory

from feature.base_feature import WeUltrarichBaseFeature
from feature.test_feature import WeUltrarichTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeUltrarichBaseFeature(),
        "test": lambda: WeUltrarichTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
